### PR TITLE
Implemented copying and pasting in the graphical editor. 

### PR DESCRIPTION
**Pull request by philip-alldredge**
Implemented copying and pasting in the graphical editor. Closes issue #1167.

### CODE REVIEW

**Comment by philip-alldredge** [comment]
@RyanMcilnay please review and testing. Of particular interest in the testing is the behavior when editing multiple diagrams and/or multiple packages and the behavior when package with statements or component type renames are used.

**Comment by philip-alldredge** [comment]
@RyanMcilnay changes pushed to correct potential issue.

File: ge/org.osate.ge/src/org/osate/ge/internal/ui/editor/actions/PasteAction.java [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+	 * Copies the clipboard contents to the destination business object and diagram element.
+	 * @return the diagram elements which were created. Does not include children of created diagram elements.
+	 */
+	private Collection<DiagramElement> copyClipboardContents(final EObject dstBoToModify,
+			final DiagramElement dstDiagramElement,
+			final DiagramModification m,
+			final ReferenceBuilderService refBuilder, final ProjectProvider projectProvider,
+			final ExtensionService extService, final NamingService namingService) {
+		// Determine the minimum coordinates from the elements whose positions will be copied
+		final Point minCoordinates = getCopiedDiagramElements().stream()
```
Comment (RyanMcilnay) [inline ge/org.osate.ge/src/org/osate/ge/internal/ui/editor/actions/PasteAction.java]: Should minCoordinates be allowed to be null?

Comment (philip-alldredge) [reply ge/org.osate.ge/src/org/osate/ge/internal/ui/editor/actions/PasteAction.java]: Added comment to code describing why it is okay for minCoordinates to be null.

File: ge/org.osate.ge/src/org/osate/ge/internal/ui/dialogs/CreateDiagramComposite.java [hunk 2]
```diff
@@ -22,13 +22,17 @@
 import org.eclipse.swt.widgets.Label;
 import org.eclipse.swt.widgets.Text;
 import org.osate.ge.internal.ui.editor.AgeDiagramEditor;
+import org.osate.ge.internal.ui.util.UiUtil;
 
 import com.google.common.collect.ImmutableCollection;
 
 /**
  * Composite for prompting the user for a name and type for a diagram. Results in a filename and diagrma type.
```
[the remark this answers is not in the document]
Comment (philip-alldredge) [reply ge/org.osate.ge/src/org/osate/ge/internal/ui/dialogs/CreateDiagramComposite.java]: Fixed

File: ge/org.osate.ge.tests/src/org/osate/ge/tests/AgeGefBot.java [hunk 3]
```diff
@@ -202,17 +276,26 @@ public void createNewProjectAndPackage(final String projectName, final String pa
...
-		// Close editor for open test
-		bot.gefEditor(packageName + ".aadl_diagram").close();
+		// Create AADL Package and close the diagram
+		createAADLPackage(projectName, packageName).close();
 
 		bot.tree().expandNode(new String[] { projectName }).getNode(packageName + ".aadl").click();
 		bot.tree().contextMenu("Open").click();
 	}
 
+	// Closest the welcome page if it is open and configures SWTBot
```
[the remark this answers is not in the document]
Comment (philip-alldredge) [reply ge/org.osate.ge.tests/src/org/osate/ge/tests/AgeGefBot.java]: Fixed

File: ge/org.osate.ge/src/org/osate/ge/internal/services/impl/DefaultClipboardService.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	private static class SimpleClipboard implements Clipboard {
+		private Object contents;
+
+		@Override
+		public Object getContents() {
+			return contents;
+		}
+
+		@Override
+		public void setContents(Object value) {
```
Comment (RyanMcilnay) [inline ge/org.osate.ge/src/org/osate/ge/internal/services/impl/DefaultClipboardService.java]: Add final to Object value

Comment (philip-alldredge) [reply ge/org.osate.ge/src/org/osate/ge/internal/services/impl/DefaultClipboardService.java]: Fixed

**Pull request merged**